CONTRIBUTING.md

```diff
@@ -1,5 +1,84 @@
+
+### Proposing a Feature
+
+In order to be respectful of the time of community contributors, we aim to discuss new potential feature and changes in GitHub issues prior to their implementation. This allows us to vet the potential utility of the change, as well as discuss how it could potentially fit into work currently being done, or a future milestone.
+
+When proposing a feature we ask that you include:
+
+- A description of the feature to add
+- Any relevant Github issues, pull requests, blog posts, and/or documentation that provides greater context
+
+## Pull Requests
+
+Your pull request should have a description of what it accomplishes. If applicable, PRs should include testing that validate correctness. In all cases the existing tests must pass.
+
+### Guidelines
+
+Before you submit your Pull Request (PR) consider the following guidelines:
+
+- Search GitHub for an open or closed PR that relates to your submission. You don't want to duplicate existing efforts.
+- Be sure that an existing issue describes the problem you're fixing, or documents the design for the feature you'd like to add. Discussing the design upfront helps to ensure that we're ready to accept your work. Make sure that this issue is referenced in your PR– this can be done by including `Closes: #[issue number]` in the PR's description.
+- Please sign our Contributor License Agreement (CLA) before sending PRs. We cannot accept code without a signed CLA. Make sure you author all contributed Git commits with email address associated with your CLA signature.
+
+### Style
+
+It is recommended that Pull Request titles follow the following format:
+
+**Type(Affected_Component): Short description of PR**
+
+#### Type:
+
+- feat(Affected_Component): A addition of a feature within the Affected Component
+- fix(Affected_Component): A bug fix within the Affected Component
+- refactor(Affected_Component): A refactor within the Affected Component
+- chore: Updates to docs, CI work, and more generally work that does not fall under the previous categories
+
+#### Affected Component:
+
+- cli
+- cdktf
+- hcl2cdk
+- provider-generator
+- examples
+
+Ensuring your PR titles follow this format helps us quickly identify the purpose of the PR and scope in which it contributes.
+
+### Pull Request Lifecycle
+
+1. You are welcome to submit an initial draft pull request for commentary before it is fully completed. It's also helpful to include comments on items you'd like feedback on or feel needs further discussion. Once you believe your pull request is ready to be merged you can set your pull request to open.
+2. When time permits, Terraform CDK team members will review your PR. From here the pull request will either be merged, or additional changed may be requested in comments. We may also have questions that we need answered about the code, due to something that needs greater clarification or just because we want to better understand your thought process.
```

Comment:
Good idea! May also be of some use to specify that there's an expectation for PR authors to get their work in agreement with the CI– with of course some help/suggestions from the team in difficult situations. Same point can be made about dealing with Merge conflicts– as well as pointing out the need for awareness and communication to avoid concurrent work in the same code region.